src/paas_charm/charm_state.py

```diff
@@ -156,42 +155,50 @@ def from_charm(  # pylint: disable=too-many-arguments
             logger.error(error_messages.long)
             raise CharmConfigInvalidError(error_messages.short) from exc
 
-        integrations = IntegrationsState.build(
-            app_name=app_name,
-            redis_uri=(integration_requirers.redis.url if integration_requirers.redis else None),
-            database_requirers=integration_requirers.databases,
-            s3_connection_info=(
-                integration_requirers.s3.get_s3_connection_info()
-                if integration_requirers.s3
-                else None
-            ),
-            saml_relation_data=(
-                saml_data.to_relation_data()
-                if (
-                    integration_requirers.saml
-                    and (saml_data := integration_requirers.saml.get_relation_data())
-                )
-                else None
-            ),
-            rabbitmq_uri=(
-                integration_requirers.rabbitmq.rabbitmq_uri()
-                if integration_requirers.rabbitmq
-                else None
-            ),
-            tracing_requirer=integration_requirers.tracing,
-            smtp_relation_data=(
-                smtp_data.to_relation_data()
-                if (
-                    integration_requirers.smtp
-                    and (smtp_data := integration_requirers.smtp.get_relation_data())
-                )
-                else None
-            ),
-        )
+        # 2025/04/15 When done ejecting IntegrationParameters, we should remove the build function
+        # and just wrap the PydanticObjects from IntegrationRequirer libs into the
+        # IntegrationState, without the build function. See integration_requirers.s3.
```

Comment:
Could we add this to the jira backlog? Usually these kind of comments live forever in the code and confuse future developers :D

Reply:
Added https://warthogs.atlassian.net/browse/ISD-3362